### PR TITLE
Larger world size + support to enable it

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description
This PR increases the size of the large world from 868x868 to 1732x1732, and includes work done to support the increased world size.

Updated configs:
* removed transition runtime v2 templates from all configs (part of UTY-1735)
* updated `default_lauch.json` to support `SimulatedPlayerCoordinator` worker type (UTY-1731)
* reduce `UnityClient` and `UnityGameLogic` checkout radius to 150 (previously 210)

We now generate the level gameobject at runtime instead of instantiating a prefab (UTY-1768). This is because the prefab asset for the new large level would be 119 MB, higher than the 100 MB file size limit set by git. As part of this:
* Remove all `FPS-Start_*` level prefabs
* Update `MapBuilder` to have an editor window
* Add helper methods to `MapBuilder` for getting world size, dimensions and generating the map
* `MapBuilderSettings` scriptable object
* Update `WorkerConnectorBase` to call `MapBuilder` inside `LoadWorld()`
* Size of generated world depends on `world_size` worker flag
	* "small" = 4 layers, "large" = 24 layers, "N" = N layers
    * world width = (num_layers * tile_width * 2) + 4, where tile_width is currently 36

#### Tests
- [x] test in editor with local deployment
- [x] test on built-out client with cloud deployment
- [x] test on mobile device with local deployment
- [x] test on mobile device with cloud deployment

**Did you remember a changelog entry?**
yes

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

@zeroZshadow @jessicafalk

### CODE REVIEW

File: workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs [hunk 1]
```diff
@@ -55,6 +35,11 @@ private void Awake()
             connectionController = GetComponent<ConnectionController>();
         }
 
+        protected override string GetWorkerType()
+        {
+            return WorkerUtils.iOSClient;
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: that can be inlined as only used once :P planned to do it and forgot

File: default_launch.json [hunk 2]
```diff
@@ -26,6 +26,21 @@
         "options": {
           "manual_worker_connection_only": true
         }
+      },
+      {
+        "layer": "SimulatedPlayerCoordinator",
+        "points_of_interest": {
```
Comment (zeroZshadow) [inline default_launch.json]: Since SimulatedPlayerCoordinator'ss don't really have a position, why not make this a more readable loadbalancer? rectangle_grid or entity_id_sharding sound like a better candidate.

File: workers/unity/Assets/Fps/Scripts/SetupLogic/MobileWorkerConnectorBase.cs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        protected override void HandleWorkerConnectionFailure(string errorMessage)
+        {
+            connectionController.OnFailedToConnect();
+        }
+
+        public override void Dispose()
+        {
+            if (levelInstance != null)
+            {
+                Destroy(levelInstance);
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/MobileWorkerConnectorBase.cs]: set to null after destroying

File: workers/unity/Assets/Fps/Scripts/SetupLogic/MobileWorkerConnectorBase.cs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            var workerSystem = Worker.World.GetExistingManager<WorkerSystem>();
+            var worldSize = Worker.Connection.GetWorkerFlag("world_size");
+
+            switch (worldSize)
+            {
+                case Small:
+                    worldLayerCount = 4;
+                    break;
+                case Large:
+                    worldLayerCount = 24;
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/MobileWorkerConnectorBase.cs]: This should not be hardcoded in a mobile specific file. Move to a Scriptable object.

File: workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerCoordinatorWorkerConnector.cs [hunk 5]
```diff
@@ -16,8 +16,7 @@ public class SimulatedPlayerCoordinatorWorkerConnector : WorkerConnectorBase
     private const string FlagDevAuthTokenId = "fps_simulated_players_dev_auth_token_id";
     private const string FlagTargetDeployment = "fps_simulated_players_target_deployment";
 
-    private static readonly Vector3 SmallWorldSize = new Vector3(148f, 100f, 148f);
-    private static readonly Vector3 LargeWorldSize = new Vector3(1156f, 100f, 1156f);
+    private static readonly Vector3 DefaultWorldBounds = new Vector3(292f, 100f, 292f);
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerCoordinatorWorkerConnector.cs]: magic hardcoded values should move to a scriptable object

File: workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerCoordinatorWorkerConnector.cs [hunk 6]
```diff
@@ -191,7 +190,12 @@ public void UnregisterLocalSimulatedPlayer(EntityId entityId, GameObject prefab)
 
     public Bounds GetWorldBounds()
     {
-        var isSmall = Worker.Connection.GetWorkerFlag("world_size") == "small";
-        return new Bounds(Worker.Origin, isSmall ? SmallWorldSize : LargeWorldSize);
+        if (GetWorldLayerCount(out var worldLayerCount))
+        {
+            var dimensions = (worldLayerCount * 72) + 4;
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerCoordinatorWorkerConnector.cs]: More magic numbers

File: workers/unity/Assets/Fps/Scripts/SetupLogic/WorkerConnectorBase.cs [hunk 7]
```diff
@@ -51,31 +48,48 @@ public override void Dispose()
...
-                workerSystem.LogDispatcher.HandleLog(LogType.Error,
-                    new LogEvent(
-                            "Invalid world_size worker flag. Make sure that it is either small, medium, or large,")
-                        .WithField("world_size", worldSize));
-                return;
+                case Small:
+                    worldLayerCount = 4;
+                    break;
+                case Large:
+                    worldLayerCount = 24;
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/WorkerConnectorBase.cs]: Duplicated code with the mobile bit

File: workers/unity/Assets/Fps/Scripts/Editor/GameLogic/WorldTiles/MapQualitySettingsInspector.cs [hunk 8]
```diff
@@ -5,7 +5,7 @@
 namespace Fps
 {
     [CustomEditor(typeof(MapQualitySettings))]
-    public class MapQualitySettingsInspector : Editor
+    public class MapQualitySettingsInspector : UnityEditor.Editor
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/Editor/GameLogic/WorldTiles/MapQualitySettingsInspector.cs]: is that change needed?

Comment (paulbalaji) [reply workers/unity/Assets/Fps/Scripts/Editor/GameLogic/WorldTiles/MapQualitySettingsInspector.cs]: yes, otherwise it complains about missing references

File: CHANGELOG.md [hunk 9]
```diff
@@ -7,16 +7,25 @@
...
+	- Update `default_launch.json` to support `SimulatedPlayerCoordinator`.
+	- Reduce checkout radius of `UnityClient` and `UnityGameLogic` workers from 210 to 150.
+- Increase the level size from 868x868 to 1732x1732.
+- Changed how level generation and `MapBuilder.cs` works.
+	- `LoadWorld()` now generates the level GameObject through `MapBuilder`, instead of instantiating it from a prefab.
 
 ### Removed
 
 - Removed the `SimulatedPlayer` worker type.
+- Remove all `FPS-Start_*` level prefabs.
```
Comment (jessicafalk) [inline CHANGELOG.md]:
```suggestion
- Removed all `FPS-Start_*` level prefabs.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -7,16 +7,25 @@
...
 ### Changed
 
 - A simulated player now connects as a regular `UnityClient` worker rather than a `SimulatedPlayer` worker.
 - In cloud deployments, `SimulatedPlayerCoordinator` workers are ran in a separate deployment to the `UnityGameLogic` workers.
     - Note that simulated players started by the coordinators still connect into the deployment with the `UnityGameLogic` workers.
+- Updated all launch configurations.
+	- Replace transition runtime v2 templates from all launch configurations.
+	- Update `default_launch.json` to support `SimulatedPlayerCoordinator`.
+	- Reduce checkout radius of `UnityClient` and `UnityGameLogic` workers from 210 to 150.
+- Increase the level size from 868x868 to 1732x1732.
```
Comment (jessicafalk) [inline CHANGELOG.md]:
```suggestion
- Increased the level size from 868x868 to 1732x1732.
```

Comment (jessicafalk) [reply CHANGELOG.md]: and the same for all the other sentences. not gonna suggest it everywhere :P

File: CHANGELOG.md [hunk 11]
```diff
@@ -7,16 +7,25 @@
...
 
 ### Changed
 
 - A simulated player now connects as a regular `UnityClient` worker rather than a `SimulatedPlayer` worker.
 - In cloud deployments, `SimulatedPlayerCoordinator` workers are ran in a separate deployment to the `UnityGameLogic` workers.
     - Note that simulated players started by the coordinators still connect into the deployment with the `UnityGameLogic` workers.
+- Updated all launch configurations.
+	- Replace transition runtime v2 templates from all launch configurations.
+	- Update `default_launch.json` to support `SimulatedPlayerCoordinator`.
+	- Reduce checkout radius of `UnityClient` and `UnityGameLogic` workers from 210 to 150.
```
Comment (jessicafalk) [inline CHANGELOG.md]:
```suggestion
	- Reduced checkout radius of `UnityClient` and `UnityGameLogic` workers from 210 to 150.
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -7,16 +7,25 @@
...
+- Added a `MapBuilderSettings` scriptable object containing the default number of layers for the small and large sized worlds.
 
 ### Changed
 
 - A simulated player now connects as a regular `UnityClient` worker rather than a `SimulatedPlayer` worker.
 - In cloud deployments, `SimulatedPlayerCoordinator` workers are ran in a separate deployment to the `UnityGameLogic` workers.
     - Note that simulated players started by the coordinators still connect into the deployment with the `UnityGameLogic` workers.
+- Updated all launch configurations.
+	- Replace transition runtime v2 templates from all launch configurations.
+	- Update `default_launch.json` to support `SimulatedPlayerCoordinator`.
```
Comment (jessicafalk) [inline CHANGELOG.md]:
```suggestion
	- Updated `default_launch.json` to support `SimulatedPlayerCoordinator`.
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -7,16 +7,25 @@
...
 - Added cloud support for Android workers.
+- Added a `MapBuilderSettings` scriptable object containing the default number of layers for the small and large sized worlds.
 
 ### Changed
 
 - A simulated player now connects as a regular `UnityClient` worker rather than a `SimulatedPlayer` worker.
 - In cloud deployments, `SimulatedPlayerCoordinator` workers are ran in a separate deployment to the `UnityGameLogic` workers.
     - Note that simulated players started by the coordinators still connect into the deployment with the `UnityGameLogic` workers.
+- Updated all launch configurations.
+	- Replace transition runtime v2 templates from all launch configurations.
```
Comment (jessicafalk) [inline CHANGELOG.md]:
```suggestion
	- Replaced transition runtime v2 templates from all launch configurations.
```

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs [hunk 14]
```diff
@@ -344,14 +361,14 @@ private void PlaceTile(Vector2Int tileCoord)
 
         private void PlaceTile(Vector2Int tileCoord, GameObject tile, float rotation)
         {
-            var tileOffset = UnitsPerTile / 2;
+            var tileOffset = mapBuilderSettings.UnitsPerTile / 2;
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: this throws a nullpointer exceptton if you don't set `mapBuilderSettings` before hitting `Generate Map`. should have a check for that in the beginning

File: workers/unity/Assets/Fps/Prefabs/iOSClientWorker.prefab [hunk 15]
```diff
@@ -41,32 +32,32 @@ Transform:
...
   m_Enabled: 1
   m_EditorHideFlags: 0
-  m_Script: {fileID: 11500000, guid: c3f6fc6158bac8a4aac46762f5dbd51d, type: 3}
+  m_Script: {fileID: 11500000, guid: f06f2bfbdf944f541b6afd42c1ce7c1c, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
---- !u!114 &114871985853100216
+  MaxConnectionAttempts: 3
+  DevelopmentAuthToken: 
+  forcedIpAddress: 172.16.117.61
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Prefabs/iOSClientWorker.prefab]: This should not be committed

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs [hunk 16]
```diff
@@ -434,5 +444,64 @@ public void Clean()
...
             }
         }
+
+        public static Vector3 GetWorldDimensions(MapBuilderSettings mapBuilderSettings, int worldLayerCount)
+        {
+            var dimensions = (worldLayerCount * mapBuilderSettings.UnitsPerTile * 2) + mapBuilderSettings.UnitsPerBlock;
+            return new Vector3(dimensions, 100f, dimensions);
+        }
+
+        public static bool GetWorldLayerCount(MapBuilderSettings mapBuilderSettings,
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: `TryGetWorldLayerCount` as it does not return the value

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs [hunk 17]
```diff
@@ -434,5 +444,64 @@ public void Clean()
...
+            return connection.GetWorkerFlag("world_size");
+        }
+
+        // Get the world size from the config, and use it to generate the correct-sized level
+        public static GameObject GenerateMap(
+            MapBuilderSettings mapBuilderSettings,
+            Transform workerTransform,
+            Connection connection,
+            string workerType,
+            WorkerSystem workerSystem)
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: This should just be an ILogDispatches, as the system isnt used

File: workers/unity/Assets/Fps/Scripts/Editor/MapBuilderVisualisationWindow.cs [hunk 18]
```diff
@@ -60,6 +62,11 @@ public void OnGUI()
                     "Different seeds produce different maps."),
                 seed);
 
+            mapBuilderSettings = (MapBuilderSettings) EditorGUILayout.ObjectField(new GUIContent(
+                    "Map Builder Settings",
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/Editor/MapBuilderVisualisationWindow.cs]: This formatting looks wrong, was this file formatted?

File: workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs [hunk 19]
```diff
@@ -25,8 +25,7 @@ public class AndroidWorkerConnector : MobileWorkerConnector, ITileProvider
         public bool ShouldConnectLocally;
         public int TargetFrameRate = 60;
 
-        public GameObject SmallLevelPrefab;
-        public GameObject LargeLevelPrefab;
+        public MapBuilderSettings MapBuilderSettings;
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs]: `[SerializeField] private`

File: workers/unity/Assets/Fps/Scripts/SetupLogic/WorkerConnectorBase.cs [hunk 20]
```diff
@@ -9,6 +9,8 @@ public abstract class WorkerConnectorBase : DefaultWorkerConnector
     {
         public int TargetFrameRate = 60;
 
+        public MapBuilderSettings MapBuilderSettings;
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/WorkerConnectorBase.cs]: `[SerializeField] private`

File: workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs [hunk 21]
```diff
@@ -21,15 +21,11 @@ public class iOSWorkerConnector : MobileWorkerConnector, ITileProvider
...
-        private const string Small = "small";
-        private const string Large = "large";
-
         public string forcedIpAddress;
 
         public int TargetFrameRate = 60;
 
-        public GameObject SmallLevelPrefab;
-        public GameObject LargeLevelPrefab;
+        public MapBuilderSettings MapBuilderSettings;
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: `[SerializeField] private`

File: workers/unity/Assets/Fps/Scripts/Editor/MapBuilderVisualisationWindow.cs (hunk 18 above)
Comment (paulbalaji) [reply workers/unity/Assets/Fps/Scripts/Editor/MapBuilderVisualisationWindow.cs]: yeah I was confused to, I did the code cleanup thing in Rider too

File: workers/unity/Assets/Fps/Scripts/SetupLogic/WorkerConnectorBase.cs (hunk 20 above)
Comment (paulbalaji) [reply workers/unity/Assets/Fps/Scripts/SetupLogic/WorkerConnectorBase.cs]: this actually needs to be protected, because the `SimulatedPlayerCoordinatorWorkerConnector` needs to be able to read the `MapBuilderSettings` to define its world bounds

File: workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs [hunk 22]
```diff
@@ -166,7 +166,7 @@ protected virtual void LoadWorld()
                 transform,
                 Worker.Connection,
                 Worker.WorkerType,
-                Worker.World.GetExistingManager<WorkerSystem>());
+                Worker.World.GetExistingManager<WorkerSystem>().LogDispatcher);
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs]: You can use `Worker.LogDispatcher` directly
